tests/python/unittest/test_subgraph_op.py

```diff
@@ -18,351 +18,448 @@
+            (network_structure_3(), ['exp', 'sin', '_Plus', 'elemwise_add', '_plus']),
+            (network_structure_3(), ['exp', 'cos', '_Plus', 'elemwise_add', '_plus']),
+            (network_structure_3(), ['exp', 'sin', '_Plus', 'elemwise_add', '_plus', 'BatchNorm']),
+            (network_structure_3(), ['exp', 'cos', '_Plus', 'elemwise_add', '_plus', 'BatchNorm']),
+            (network_structure_3(), ['exp', 'BatchNorm']),
+            (network_structure_3(), ['BatchNorm']),
+            (network_structure_4(), ['exp']),
+            (network_structure_5(), ['_plus', '_Plus', 'elemwise_add']),
+            (network_structure_6(), []),
+            (network_structure_6(), [mx.sym.sin.__name__]),
+            (network_structure_6(), [mx.sym.Convolution.__name__]),
+            (network_structure_6(), [mx.sym.sin.__name__, mx.sym.Convolution.__name__]),
+            (network_structure_7(), ['sin', 'elemwise_add', '_plus', '_Plus'])
+            ]
+
+def check_subgraph_exe1(sym, subgraph_backend, op_names):
+    """Use the partitioned sym to simple_bind an executor and compare the outputs
+    with those of the original executor"""
+    out = SymbolHandle()
+    check_call(_LIB.MXBuildSubgraphByOpNames(sym.handle, c_str(subgraph_backend), mx_uint(len(op_names)),
+                                              c_str_array(op_names), ctypes.byref(out)))
+
+    partitioned_sym = Symbol(out)
+    assert partitioned_sym.list_inputs() == sym.list_inputs()
+    assert partitioned_sym.list_arguments() == sym.list_arguments()
+    assert partitioned_sym.list_auxiliary_states() == sym.list_auxiliary_states()
+    exe = sym.simple_bind(ctx=mx.current_context(), grad_req='null')
+    partitioned_exe = partitioned_sym.simple_bind(ctx=mx.current_context(), grad_req='null')
+    input_names = sym.list_inputs()
+    for name in input_names:
+        if name in exe.arg_dict:
+            exe.arg_dict[name][:] = mx.nd.random.uniform(shape=exe.arg_dict[name].shape)
+            partitioned_exe.arg_dict[name][:] = exe.arg_dict[name]
+        else:
+            assert name in exe.aux_dict
+            exe.aux_dict[name][:] = mx.nd.random.uniform(shape=exe.aux_dict[name].shape)
+            partitioned_exe.aux_dict[name][:] = exe.aux_dict[name]
+    exe.forward()
+    partitioned_exe.forward()
+    assert len(exe.outputs) == len(partitioned_exe.outputs)
+    for i in range(len(exe.outputs)):
+        assert_almost_equal((exe.outputs[i] - partitioned_exe.outputs[i]).abs().sum().asnumpy(),
+                            np.zeros(shape=(1,)))
+
+def check_subgraph_exe2(sym, subgraph_backend, op_names):
+    """Use env var MXNET_SUBGRAPH_BACKEND=default to trigger graph partitioning in simple_bind
+    and compare results of the partitioned sym and the original sym."""
+    def get_executor(sym, subgraph_backend=None, op_names=None, original_exec=None):
+        if subgraph_backend is not None:
+            os.environ['MXNET_SUBGRAPH_BACKEND'] = subgraph_backend
```

Comment:
nit: there's a decorator somewhere to restore the environment using with, this is not really safe if there's an exception and the environment is left in a bad state.

Reply:
This is old code, we're not adding it in this PR just refactoring it. So we'll ignore the nit for this PR